units/patch_intent.go

```diff
@@ -483,6 +493,46 @@ func (j *patchIntentProcessor) getPreviousPatchDefinition(project *model.Project
 	return res, nil
 }
 
+func getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant []string, displayTasksInProjectVariant []string, vt patch.VariantTasks, version string) ([]string, []patch.DisplayTask, error) {
+	failedTasks, err := task.FindAll(db.Query(task.FailedExecutionTasksByVersionAndBV(version, vt.Variant)))
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "error querying for failed tasks from previous patch")
+	}
+
+	failedTaskNames := make([]string, 0, len(failedTasks))
+	for _, failedTask := range failedTasks {
+		failedTaskNames = append(failedTaskNames, failedTask.DisplayName)
+	}
+	// I want the subset of vt.tasks that exist in tasksForVariant
+	tasks := utility.StringSliceIntersection(tasksInProjectVariant, failedTaskNames)
+
+	var displayTasks []patch.DisplayTask
+	failedDisplayTasks, err := task.FindAll(db.Query(task.FailedDisplayTasksByVersionAndBV(version, vt.Variant)))
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "error querying for failed display tasks from previous patch")
+	}
+	for _, dt := range vt.DisplayTasks {
```

Comment:
I'm having trouble parsing this logic from the logic at 524. Is it possible to avoid this double for loop? If we convert failedDisplayTasks here into a string list instead of a display task list in the big for loop I suggested in the previous comment, could we just pass these things into getPreviousTasksAndDisplayTasks instead?

Reply:
Good point, I've converted failed display tasks and execution tasks to string arrays and now passing them into getPreviousTasksAndDisplayTasks